liblumen_alloc/src/erts/process/trace/utils.rs

```diff
@@ -0,0 +1,264 @@
+
+            let module: Atom = module_tt.try_into().map_err(|_| ())?;
+            let function: Atom = function_tt.try_into().map_err(|_| ())?;
+            let mut symbol = match arity_or_args {
+                TypedTerm::SmallInteger(i) => Symbolication {
+                    mfa: Some(ModuleFunctionArity {
+                        module,
+                        function,
+                        arity: i.try_into().unwrap(),
+                    }),
+                    filename: None,
+                    line: None,
+                },
+                TypedTerm::Nil => Symbolication {
+                    mfa: Some(ModuleFunctionArity {
+                        module,
+                        function,
+                        arity: 0,
+                    }),
+                    filename: None,
+                    line: None,
+                },
+                TypedTerm::List(list) => Symbolication {
+                    mfa: Some(ModuleFunctionArity {
+                        module,
+                        function,
+                        arity: list.iter().count() as u8,
+                    }),
+                    filename: None,
+                    line: None,
+                },
+                _ => return Err(()),
+            };
+
+            match meta_tt {
+                // No location metadata
+                TypedTerm::Nil => Ok(symbol),
+                TypedTerm::List(list) => {
+                    // Get file metadata and convert to path
+                    let file = list
+                        .keyfind(0, Atom::str_to_term("file"))
+                        .ok()
+                        .unwrap_or(None);
+                    if let Some(TypedTerm::List(chars)) = file.map(|t| t.decode().unwrap()) {
+                        symbol.filename = Some(PathBuf::from(format!("{}", chars.as_ref())));
+                    }
+
+                    // Get line metadata and convert to u32
+                    let line = list
+                        .keyfind(0, Atom::str_to_term("line"))
+                        .ok()
+                        .unwrap_or(None);
+                    if let Some(TypedTerm::SmallInteger(i)) = line.map(|t| t.decode().unwrap()) {
+                        let i: u32 = i.try_into().unwrap();
+                        symbol.line = Some(i);
+                    }
+
+                    Ok(symbol)
+                }
+                // This technically is a bug, but its non-essential info, so ignore it
+                _ => Ok(symbol),
+            }
+        } else {
+            Err(())
+        }
+    }
+}
+
+// Minimum frame size is a 4 element tuple with all immediates (i.e. nil for the location info)
+pub const MIN_FRAME_SIZE: usize = 5 * mem::size_of::<Term>();
+// Base frame size is a 4 element tuple, 2 cons cells, and 2 2-element tuples,
+// and 1 character list of up to 120 chars of which each char requires a cons cell
+pub const MAX_FILENAME_LEN: usize = 120;
```

Comment:
Why a max of 120 characters?  We know Window's 256 character path limit causes problems, so why 120 here?

Reply:
The trace code trims the paths down to this limit (leaving the tail of the path intact), so we're not discarding useful information, just excess path information that probably shouldn't be there anyway (i.e. absolute paths). The real reason is to avoid allocating enormous amounts of memory for a backtrace just due to long paths being present in the debug info. If paths were stored as binary terms and not lists, this wouldn't be worth the effort, but cons cells are so wasteful in terms of memory that placing a hard limit on them is well worth it for both performance and memory savings.

Reply:
Honestly every time I think about traces using lists for paths it makes me mad, I wonder if that detail is depended on in enough places to prevent getting that changed in OTP

Reply:
Eh, leaving the tail is probably good enough.  It's still better than lldb only showing the basename by default.